### PR TITLE
Pract.07: Add 09 | Mihail Yordanov

### PR DESCRIPTION
**Pull request by BojeMoii**

### CODE REVIEW

**Comment by Justsvetoslavov** [comment]
PR description is missing

File: SI/Task09.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+
+int Broqch(char* arr)
```
Comment (GeorgiTerziev02) [inline SI/Task09.cpp:3]: Change the name of the function, const char* arr - you are not modifying the char[]

File: SI/Task09.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+int Index(char* text, char* word)
+{
+	int pom = -1;
+	for (int i = 0; i < Broqch(text); i++)
+	{
+		if (text[i] == word[0])
+		{
+			bool isTrue = true;
+			for (int j = 0; j < Broqch(word); j++)
```
Comment (GeorgiTerziev02) [inline SI/Task09.cpp:21]: Big mistake - fix the for loop - you are calculating the length of the char[] **after every iteration** of the for loop
=> size_t length = myStrLen(word) and then use length in the condition of the for loop

File: SI/Task09.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+			{
+				if (text[i + j] != word[j])
+				{
+					isTrue = false;
+				}
```
Comment (GeorgiTerziev02) [inline SI/Task09.cpp:26]: What if i + j is bigger or equal to the length of the text?

File: SI/Task09.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	{
+		br++;
+	}
+	return br;
+}
+
+int Index(char* text, char* word)
+{
+	int pom = -1;
+	for (int i = 0; i < Broqch(text); i++)
```
Comment (GeorgiTerziev02) [inline SI/Task09.cpp:16]: Big mistake - fix the for loop - you are calculating the length of the char[] after every iteration of the for loop
=> size_t length = myStrLen(text) and then use length in the condition of the for loop

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**